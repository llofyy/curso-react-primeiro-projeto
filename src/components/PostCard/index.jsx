import React from 'react';
import Types from 'prop-types';

import './styles.css';

export default function PostCard({ title, body, cover }) {
  return (
    <div className="post">
      <img src={cover} alt={title} />
      <div className="post-content">
        <h1>{title}</h1>
        <p>{body}</p>
      </div>
    </div>
  );
}

PostCard.propTypes = {
  title: Types.string.isRequired,
  body: Types.string.isRequired,
  cover: Types.string.isRequired,
};
