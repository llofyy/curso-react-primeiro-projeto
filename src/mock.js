export const PostCardProps = {
    title: 'Title 1',
    body: 'Alguma descrição',
    cover: 'https://github.com/llofyy.png'
}