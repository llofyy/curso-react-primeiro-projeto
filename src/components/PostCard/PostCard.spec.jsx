import { render, screen } from '@testing-library/react';
import PostCard from '.';
import { PostCardProps } from '../../mock';

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard { ...PostCardProps } />);

        expect(screen.getByRole('img', { name: PostCardProps.title})).toHaveAttribute('src', PostCardProps.cover);
        expect(screen.getByRole('heading', { name: PostCardProps.title})).toBeInTheDocument();
        expect(screen.getByText(PostCardProps.body)).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard { ...PostCardProps } />);
        expect(container.firstChild).toMatchSnapshot();
    });
});  