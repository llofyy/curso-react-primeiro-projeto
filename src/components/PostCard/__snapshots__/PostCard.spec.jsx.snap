// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="https://github.com/llofyy.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Title 1
    </h1>
    <p>
      Alguma descrição
    </p>
  </div>
</div>
`;
